FsFilter1/FsFilter1.cpp

```diff
@@ -74,6 +84,10 @@ FLT_PREOP_CALLBACK_STATUS MiniPreWrite(PFLT_CALLBACK_DATA Data, PCFLT_RELATED_OB
 		status = FltParseFileNameInformation(FileNameInfo);
 		if (NT_SUCCESS(status)) {
 			RtlCopyMemory(FileName, FileNameInfo->Name.Buffer, FileNameInfo->Name.MaximumLength);
+			if (wcsncmp(FileName, drive_filter.Buffer, wcslen(drive_filter.Buffer)) != 0) {
+				KdPrint(("%ws is filtered out. Does not match %ws. Returning back\n", FileName, drive_filter.Buffer));
```

Comment:
Should we call FltReleaseFileNameInformation in this block before returning?

Reply:
And also I think we should add the same logic to MiniPreCreate also.

Reply:
Addressed